framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -135,18 +139,41 @@ public DefaultTypeArgumentInference(AnnotatedTypeFactory typeFactory) {
         }
 
         final Set<TypeVariable> targets = TypeArgInferenceUtil.methodTypeToTargets(methodType);
-        final Map<TypeVariable, AnnotatedTypeMirror> inferredArgs =
-                infer(typeFactory, argTypes, assignedTo, methodElem, methodType, targets, true);
-
-        if (showInferenceSteps) {
-            checker.message(Kind.NOTE, "  after infer: %s\n", inferredArgs);
-        }
-
-        handleNullTypeArguments(
-                typeFactory, methodElem, methodType, argTypes, assignedTo, targets, inferredArgs);
-
-        if (showInferenceSteps) {
-            checker.message(Kind.NOTE, "  after handleNull: %s\n", inferredArgs);
+        Map<TypeVariable, AnnotatedTypeMirror> inferredArgs;
+        if (assignedTo == null && TreeUtils.getAssignmentContext(pathToExpression) != null) {
+            inferredArgs = new LinkedHashMap<>();
```

Comment:
Can you add a comment for what the logic for the condition is? I can see the check for `assignedTo == null`, but why the condition if the assignment context is non-null? Should that also be equal to null?